app/components/Sidebar/components/TrashLink.js

```diff
@@ -0,0 +1,66 @@
+// @flow
+import { observer } from "mobx-react";
+import { TrashIcon } from "outline-icons";
+import * as React from "react";
+import { useState } from "react";
+import { useDrop } from "react-dnd";
+import { useTranslation } from "react-i18next";
+import DocumentDelete from "scenes/DocumentDelete";
+import Modal from "components/Modal";
+import useStores from "../../../hooks/useStores";
+import SidebarLink from "./SidebarLink";
+
+function TrashLink({ documents }) {
+  const { policies } = useStores();
+  const { t } = useTranslation();
+  const [showDeleteModal, setShowDeleteModal] = useState(true);
+  const [document, setDocument] = useState();
+
+  const handleModalClose = React.useCallback(() => {
+    setDocument(undefined);
+    setShowDeleteModal(false);
+  }, []);
+
+  const [{ isDocumentDropping }, dropToTrashDocument] = useDrop({
+    accept: "document",
+    drop: async (item, monitor) => {
+      setDocument(documents.get(item.id));
+      setShowDeleteModal(true);
+    },
+    canDrop: (item, monitor) => {
+      console.log(policies.abilities(item.id));
+      return policies.abilities(item.id).delete;
+    },
+    collect: (monitor) => ({
+      isDocumentDropping: monitor.isOver(),
+    }),
+  });
+
+  return (
+    <>
+      <div ref={dropToTrashDocument}>
+        <SidebarLink
+          to="/trash"
+          icon={<TrashIcon color="currentColor" open={isDocumentDropping} />}
+          exact={false}
+          label={t("Trash")}
+          active={documents.active ? documents.active.isDeleted : undefined}
+          isActiveDrop={isDocumentDropping}
+        />
+      </div>
+      {document && (
+        <Modal
+          title={t("Delete {{ documentName }}", {
+            documentName: document.noun,
+          })}
+          onRequestClose={handleModalClose}
+          isOpen={showDeleteModal}
+        >
+          <DocumentDelete document={document} onSubmit={handleModalClose} />
+        </Modal>
+      )}
```

Comment:
This logic to show modal while dropping works in chrome but not in firefox. Could you help me how to debug this kind of bug?

Reply:
@tommoor

Reply:
Will look at it, one note here is that I don't think we need state for the delete modal bool AND the document, we can probably just use the document being in state to know if to show the modal?

Reply:
It's actually working in Firefox for me 🤔

Reply:
It's not even throwing any error or warning.

Reply:
https://user-images.githubusercontent.com/380914/125659527-2c7f5a85-e877-4f14-8ce9-619fe6ffd32e.mp4

Reply:
Great! Now we can narrow down this to `firefox` on `ubuntu`. It's one of those "works on my machine" bugs xD

Reply:
I tried different changes, but nothing solved the issue. The lead, for now, is that the issue is with the `Modal` component, not the `DocumentDelete`.

Reply:
What version of Firefox? I'm on `89.0.2`

Reply:
Me too 89.0.2 (64-bit)

Reply:
I think it's down to you to debug as I still can't repro, try putting logs in the relevant drag/drop callbacks to see where it isn't getting to

Reply:
No luck finding the cause. I used browser debugging tools, and it feels like firefox is gulping the change 😞

Reply:
Does the icon animate on firefox like it's _going_ to work? Or no feedback at all?

Reply:
It does animate.